### PR TITLE
Refactors NOREACT flag; minor cig refactor

### PR DESCRIPTION
**Pull request by coiax**
The reagents datum now has its own flags, which currently includes
REAGENT_NOREACT, which functions in the same way. Shouldn't be touched
directly, use set_reacting(bool) to modify it, as modification also adds
or removes the reagents datum from the SSobj.processing list.

Also refactors cigs a little, adds a Destroy(), uses the open_flame()
proc for the hotspot exposure.

### CODE REVIEW

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 1]
```diff
@@ -165,46 +177,49 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+		var/turf/T = get_turf(src)
+		T.visible_message(flavor_text)
+	SSobj.processing |= src
+
+	//can't think of any other way to update the overlays :<
+	if(ismob(loc))
+		var/mob/M = loc
+		M.update_inv_wear_mask()
+		M.update_inv_l_hand()
+		M.update_inv_r_hand()
```
Comment (phil235) [inline code/game/objects/items/weapons/cigs_lighters.dm]: did you make any change here or is it just an indentation change?

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 2]
```diff
@@ -165,46 +177,49 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
+		var/datum/effect_system/reagents_explosion/e = new()
+		e.set_up(round(reagents.get_reagent_amount("welding_fuel") / 5, 1), get_turf(src), 0, 0)
+		e.start()
+		if(ismob(loc))
+			var/mob/M = loc
+			M.unEquip(src, 1)
+		qdel(src)
+		return
+	// allowing reagents to react after being lit
+	reagents.set_reacting(TRUE)
```
Comment (coiax) [inline code/game/objects/items/weapons/cigs_lighters.dm]: This is the only change, from the flag to the proc call

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 3]
```diff
@@ -165,46 +177,49 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
-		item_state = icon_on
-		if(flavor_text)
-			var/turf/T = get_turf(src)
-			T.visible_message(flavor_text)
-		SSobj.processing |= src
+	if(src.lit)
+		return
 
-		//can't think of any other way to update the overlays :<
+	src.lit = TRUE
```
Comment (coiax) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Oh, and a 1 to a TRUE

**Comment by pudl** [comment]
good change

File: code/game/objects/items/weapons/cigs_lighters.dm (hunk 3 above)
Comment (Jordie0608) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Drop the `src.`

File: code/game/objects/items/weapons/cigs_lighters.dm [hunk 4]
```diff
@@ -165,46 +177,49 @@ CIGARETTE PACKETS ARE IN FANCY.DM
...
-			return
-		flags &= ~NOREACT // allowing reagents to react after being lit
-		reagents.handle_reactions()
-		icon_state = icon_on
-		item_state = icon_on
-		if(flavor_text)
-			var/turf/T = get_turf(src)
-			T.visible_message(flavor_text)
-		SSobj.processing |= src
+	if(src.lit)
```
Comment (Jordie0608) [inline code/game/objects/items/weapons/cigs_lighters.dm]: Same here

**Pull request merged**